apollo-router/src/query_planner/fetch.rs

```diff
@@ -298,18 +298,43 @@ impl FetchNode {
         paths: HashMap<Path, usize>,
         response: graphql::Response,
     ) -> (Value, Vec<Error>) {
-        let mut errors: Vec<Error> = response
-            .errors
-            .into_iter()
-            .map(|error| Error {
-                locations: error.locations,
-                path: error.path.map(|path| current_dir.join(path)),
-                message: error.message,
-                extensions: error.extensions,
-            })
-            .collect();
+        let mut inverted_paths: HashMap<usize, Vec<&Path>> = HashMap::new();
+        for (path, index) in paths.iter() {
```

Comment:
Nit, possibly this section could benefit from comments. The logic seems a bit tricky, and it took a while for me to figure out what it was doing.

Reply:
right, I added more comments, the paths manipulations are confusing otherwise